plugin/example_config.toml

```diff
@@ -0,0 +1,96 @@
+###############################################################################
+###                      Plugin system configuration                        ###
+###############################################################################
+
+[plugins]
+
+# turn the plugin system, as a whole, on or off
+on = true
+
+# List of plugin names to enable from the plugin/plugins/*
+enabled = ["kafka"]
+
+# The directory to load non-preloaded plugins from; defaults $GOPATH/src/github.com/cosmos/cosmos-sdk/plugin/plugins
+dir = ""
```

Comment:
These configuration pieces should be in a `Plugins` field in the `BaseConfig` (defined in `server/config/config.go`). They should also have corresponding entries in the config file template defined in `server/config/toml.go`.

Also, since each plugin will require its own set of custom configuration parameters, I it'd be best to have a separate `plugins.toml` file for the specific configuration of a plugin (or even a file for each plugin). That is, this `[plugins]` section would go in `app.toml` but the other configuration pieces (e.g. `plugins.streaming.file` or `plugins.streaming.kafka`) would go in separate config files.

The reason I say all this, is that the `app.toml` can be updated programmatically, but it uses the `BaseConfig` struct template (defined in that `toml.go` file) to do so. If there are custom fields added to the file, those will be stomped on during such an update. Because of the nature of plugins, the `BaseConfig` struct cannot know all the configuration options available for any plugin. That then also applies to the `app.toml` config file. While you can put things in there manually that can be read and used, as far as I know, the templating system doesn't allow for custom maps like what would be needed for each plugin.

Basically, in `app.toml` we would configure the app to enable/use a plugin. But the config for that plugin should go in a separate file.

Maybe the `enabled` field could be renamed to `config_files` that point to the config files of each plugin.